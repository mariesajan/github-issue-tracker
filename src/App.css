.spacing {
  margin: 20px 10px;
}
label, button {
  margin: 10px;
}
.center {
  text-align: center;
}
.error {
  color: red;
}