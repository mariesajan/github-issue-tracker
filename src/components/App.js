import React from 'react';
import styled from 'styled-components';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import RepositoryList from './RepositoryList';
import EnterKey from './EnterKey';

const StyledRepositoryWrapper = styled.div`
  display: flex;
  justify-content: flex-start;
  flex-direction: column;
  border: 1px solid lightgray;
  border-radius: 10px;
  margin: auto;
  max-width: 1200px;
`;


const RepositoryWrapper = () => {
    return (
        <React.Fragment>
            <StyledRepositoryWrapper>
                <h2 className="center">Repository Details</h2>
                <RepositoryList />
            </StyledRepositoryWrapper>
        </React.Fragment>
    );
}

function App() {
  return (
      <Router>
          <div>
            <h1 className="center">Github Issue Tracker</h1>
            
              <Switch>
                <Route path="/" exact component={EnterKey} />
                <Route path="/repositories" component={RepositoryWrapper} />
              </Switch>
        </div>
      </Router>
  );
}

export default App;
