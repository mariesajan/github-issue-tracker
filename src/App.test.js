import React from 'react';
import renderer from 'react-test-renderer';
import App from './components/App';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('App EnterKey Component', () => {
  const initialState = {
    isError: false,
    selectedRepo: '', 
    owner: '', 
    apiKey: '', 
    repositoryDetails: [], 
    issueDetails: [] 
};
  const mockStore = configureStore();
  let store, wrapper;
  it('matches the snapshot', () => {
    store = mockStore(initialState)
    const tree = renderer.create(<Provider store={store}><App /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
 
});





