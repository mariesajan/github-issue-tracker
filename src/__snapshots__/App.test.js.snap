// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App EnterKey Component matches the snapshot 1`] = `
<div>
  <h1
    className="center"
  >
    Github Issue Tracker
  </h1>
  <div
    className="center"
  >
    <label>
      Enter the Key:
    </label>
    <input
      onChange={[Function]}
      type="text"
    />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
    
  </div>
</div>
`;
